HyperVExtension/src/HyperVExtension/CommunicationWithGuest/ApplyConfigurationOperation.cs

```diff
@@ -4,43 +4,46 @@
 using System.ComponentModel;
 using System.Globalization;
 using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
 using System.Threading;
 using HyperVExtension.HostGuestCommunication;
+using HyperVExtension.Models;
 using HyperVExtension.Providers;
 using Microsoft.Windows.DevHome.SDK;
 using Windows.Foundation;
+using Windows.Storage;
+using Windows.Storage.Streams;
 using Windows.Win32.Foundation;
-
 using SDK = Microsoft.Windows.DevHome.SDK;
 
 namespace HyperVExtension.CommunicationWithGuest;
 
-internal sealed class ApplyConfigurationOperation : IApplyConfigurationOperation, IDisposable
+public sealed class ApplyConfigurationOperation : IApplyConfigurationOperation, IDisposable
 {
-    private readonly IComputeSystem _computeSystem;
+    private readonly HyperVVirtualMachine _virtualMachine;
     private readonly CancellationTokenSource _cancellationTokenSource = new();
+
     private bool _disposed;
 
-    public ApplyConfigurationOperation(IComputeSystem computeSystem)
+    public string Configuration { get; private set; } = string.Empty;
+
+    public ApplyConfigurationOperation(HyperVVirtualMachine virtualMachine, string configuration)
     {
-        _computeSystem = computeSystem;
+        _virtualMachine = virtualMachine;
+        Configuration = configuration;
     }
 
-    public ApplyConfigurationOperation(IComputeSystem computeSystem, Exception result, string? resultDescription = null)
+    public ApplyConfigurationOperation(HyperVVirtualMachine virtualMachine, Exception result, string? resultDescription = null)
     {
-        _computeSystem = computeSystem;
-        CompletionStatus = new SDK.ApplyConfigurationResult(
-            result,
-            resultDescription,
-            null,
-            null);
+        _virtualMachine = virtualMachine;
+        CompletionStatus = new SDK.ApplyConfigurationResult(result, result.Message, result.Message);
     }
 
     public CancellationToken CancellationToken => _cancellationTokenSource.Token;
 
-    public event TypedEventHandler<IComputeSystem, SDK.ApplyConfigurationResult>? Completed;
+    public event TypedEventHandler<IApplyConfigurationOperation, ApplyConfigurationActionRequiredEventArgs> ActionRequired = (s, e) => { };

```

Comment:
Any particular reason why to initialize these events to a dummy call?
We check if it's null when we invoke it. Specifically for ActionRequired we'll need to check if it's null when we start the operation. And if it is we should fail. Also we should check that at the time of the invokation and fail if there is event handler because nobody will respond to "show adaptive card request". (Yo don't need to implement that in this PR).

Reply:
I just find it cleaner to invoke when I initiate the event this way. No other reason. doing `ActionRequired(this, args)` versus `ActionRequired?.Invoke(this, args)`

Reply:
I see it as personal religion in this case. ?Invoke is not much code. OTOH with a dummy event handler it’s:
1.	Loss of functionality: can’t check for null when needed (and we need it here).
2.	Waste of resource and likely a small perf hit calling extra even handler
To me, as an OS developer, that overweighs a few characters in invocation code. (I wish C# had a way to do check for null internally for events, so we wouldn’t need to use ?Invoke or a dummy handler)